### PR TITLE
[PLATFORM-1187] Redirect back to the creator after successful login

### PR DESCRIPTION
**Pull request by mondoreale**
Instead of creating products directly in `CreateProductModal` we now do it in `NewProductPage` component. It's a page-like component that comes with an endpoint – `/core/products/new`. This way if the product creation fails for a 403/401-reason the app takes us to the login page and, after successful login, takes us back to the product creator – plays really well with the interceptor logic we have for general unauthorized access.

If the backend creation fails for a non-auth reason the user is gonna see the generic error page instead of previous nothing.

It's subjective, of course, but I feel like this way is also much cleaner.

### CODE REVIEW

File: app/src/marketplace/components/NewProductPage.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        postEmptyProduct(sanitizedType(type))
+            .then(({ id }) => {
+                if (isMounted()) {
+                    dispatch(replace(routes.editProduct({
+                        id,
+                    })))
+                }
+            })
+            .catch(setError)
```
Comment (timoxley) [inline app/src/marketplace/components/NewProductPage.jsx]: Hm, this also catches errors in the `then` e.g. the `dispatch` reducer. Seems unlikely to happen but not sure such errors should be handled by this `.catch`. Perhaps use second argument to `.then`? I guess this is just ported from the existing code.

File: app/src/marketplace/components/NewProductPage.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+const UnstyledNewProductPage = ({ className, location: { search } }: Props) => {
+    const dispatch = useDispatch()
+
+    const isMounted = useIsMounted()
+
+    const [error, setError] = useState()
+
+    useEffect(() => {
+        const { type } = qs.parse(search)
```
Comment (timoxley) [inline app/src/marketplace/components/NewProductPage.jsx]: would explicit handling for no/unknown type be useful?

File: app/src/marketplace/components/NewProductPage.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        id,
+                    })))
+                }
+            })
+            .catch(setError)
+    }, [dispatch, isMounted, search])
+
+    useEffect(() => {
+        if (error) {
+            throw error
```
Comment (timoxley) [inline app/src/marketplace/components/NewProductPage.jsx]: 👍

File: app/src/marketplace/components/NewProductPage.jsx (hunk 2 above)
Comment (mondoreale) [reply app/src/marketplace/components/NewProductPage.jsx]: Well, the "create product modal" can't be linked to. That'd be my idea: no/unknows => open the selection tool. Right now we default no/unknown to `NORMAL`. Any ideas?

File: app/src/marketplace/components/NewProductPage.jsx (hunk 1 above)
Comment (mondoreale) [reply app/src/marketplace/components/NewProductPage.jsx]: Done in e0fbc69.

File: app/src/marketplace/components/NewProductPage.jsx (hunk 2 above)
Comment (mondoreale) [reply app/src/marketplace/components/NewProductPage.jsx]: I'mma roll with that. We can handle the edge cases later if needed.

**Pull request merged**